### PR TITLE
JIT: update `fgReplaceJumpTarget` to maintain pred lists

### PR DESCRIPTION
**Pull request by AndyAyersMS**
This is used some early phases; make it pred list aware.

Contributes to #80193.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@BruceForstall PTAL
cc @dotnet/jit-contrib 

Seems like we also might be able to use this in place of `fgReplaceSwitchJumpTarget` but will leave that for a follow-up.

File: src/coreclr/jit/fgbasic.cpp [hunk 1]
```diff
@@ -559,11 +557,17 @@ void Compiler::fgReplaceJumpTarget(BasicBlock* block, BasicBlock* newTarget, Bas
         case BBJ_ALWAYS:
         case BBJ_EHCATCHRET:
         case BBJ_EHFILTERRET:
-        case BBJ_LEAVE: // This function will be called before import, so we still have BBJ_LEAVE
+        case BBJ_LEAVE: // This function can be called before import, so we still have BBJ_LEAVE
```
Comment (BruceForstall) [inline src/coreclr/jit/fgbasic.cpp:560]: The changes in this function look fine, but also it seems weird BBJ_NONE/etc. case don't assert unreached.

**Review by BruceForstall — APPROVED** [review]

File: src/coreclr/jit/fgbasic.cpp (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/jit/fgbasic.cpp:560]: Let me fix this.

It's also a bit odd that even `block` has a reasonable jump kind this method can silently do nothing if `block` doesn't jump to `oldTarget`, but there are places we rely on this behavior right now.

**Pull request merged**